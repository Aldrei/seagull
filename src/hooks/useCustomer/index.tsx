import React, { PropsWithChildren, createContext } from 'react'
import { FaWhatsapp } from 'react-icons/fa'
import { IContactData, ICustomer, ICustomerData, IMenuData } from './types'

const useCustomer = (): ICustomer => {
  /**
   * TODO:
   *
   *  - Make the request to get data customer.
   *    But for now return by .env: .
   *
   */
  const customerData: ICustomerData = {
    domain: process.env.NEXT_PUBLIC_CUSTOMER_DOMAIN,
    name: process.env.NEXT_PUBLIC_CUSTOMER_NAME,
    abbr: process.env.NEXT_PUBLIC_CUSTOMER_ABBR,
    logo: process.env.NEXT_PUBLIC_CUSTOMER_LOGO,
    times: process.env.NEXT_PUBLIC_CUSTOMER_TIMES,
    adornment: process.env.NEXT_PUBLIC_CUSTOMER_ADORNMENT,
    limitPhotosShow: 4,
    recaptcha: {
      api_public_key: process.env.NEXT_PUBLIC_GOOGLE_RECAPTHA,
    },
    broker_1_picture: process.env.NEXT_PUBLIC_CUSTOMER_BROKER_1_PICTURE,
    social: {
      instagram: {
        title: process.env.NEXT_PUBLIC_INSTAGRAM_TITLE,
        desc: process.env.NEXT_PUBLIC_INSTAGRAM_DESC,
        link: process.env.NEXT_PUBLIC_INSTAGRAM_LINK,
        link_device: process.env.NEXT_PUBLIC_INSTAGRAM_DEVICE_LINK,
        alt: process.env.NEXT_PUBLIC_INSTAGRAM_ALT,
      },
      facebook: {
        title: process.env.NEXT_PUBLIC_FACEBOOK_TITLE,
        desc: process.env.NEXT_PUBLIC_FACEBOOK_DESC,
        link: process.env.NEXT_PUBLIC_FACEBOOK_LINK,
        link_device: process.env.NEXT_PUBLIC_FACEBOOK_DEVICE_LINK,
        alt: process.env.NEXT_PUBLIC_FACEBOOK_ALT,
      },
      linkedin: {
        title: process.env.NEXT_PUBLIC_LINKEDIN_TITLE,
        desc: process.env.NEXT_PUBLIC_LINKEDIN_DESC,
        link: process.env.NEXT_PUBLIC_LINKEDIN_LINK,
        link_device: process.env.NEXT_PUBLIC_LINKEDIN_DEVICE_LINK,
        alt: process.env.NEXT_PUBLIC_LINKEDIN_ALT,
      },
      whatsapp: {
        title: process.env.NEXT_PUBLIC_WHATSAPP_TITLE,
        desc: process.env.NEXT_PUBLIC_WHATSAPP_DESC,
        link: process.env.NEXT_PUBLIC_WHATSAPP_LINK,
        link_device: process.env.NEXT_PUBLIC_WHATSAPP_DEVICE_LINK,
        alt: process.env.NEXT_PUBLIC_WHATSAPP_ALT,
      },
      phone_1: {
        title: process.env.NEXT_PUBLIC_PHONE_1_TITLE,
        desc: process.env.NEXT_PUBLIC_PHONE_1_DESC,
        link: process.env.NEXT_PUBLIC_PHONE_1_LINK,
        link_device: process.env.NEXT_PUBLIC_PHONE_1_DEVICE_LINK,
        alt: process.env.NEXT_PUBLIC_PHONE_1_ALT,
      },
      email_1: {
        title: process.env.NEXT_PUBLIC_EMAIL_1_TITLE,
        desc: process.env.NEXT_PUBLIC_EMAIL_1_DESC,
        link: process.env.NEXT_PUBLIC_EMAIL_1_LINK,
        link_device: process.env.NEXT_PUBLIC_EMAIL_1_DEVICE_LINK,
        alt: process.env.NEXT_PUBLIC_EMAIL_1_ALT,
      },
    },
    layout: {
      nav: 'layout_nav_1',
    },
    infos: {
      times: process.env.NEXT_PUBLIC_CUSTOMER_INFO_TIMES,
    },
  }

  const menuData: IMenuData[] = []
  const contactList: IContactData[] = []

  if (customerData.layout.nav === 'layout_nav_1') {
    if (process.env.NEXT_PUBLIC_WHATSAPP_DESC)
      menuData.push({
        title: process.env.NEXT_PUBLIC_WHATSAPP_TITLE || '',
        desc: process.env.NEXT_PUBLIC_WHATSAPP_DESC || '',
        link_external: process.env.NEXT_PUBLIC_WHATSAPP_LINK_EXTERNAL || '',
        icon: <FaWhatsapp />,
      })

    menuData.push({
      title: 'Imóveis',
      desc: 'Imóveis',
      link: '/c',
    })

    if (process.env.NEXT_PUBLIC_PHONE_1_DESC)
      contactList.push({
        desc: process.env.NEXT_PUBLIC_PHONE_1_DESC,
      })

    if (process.env.NEXT_PUBLIC_PHONE_2_DESC)
      contactList.push({
        desc: process.env.NEXT_PUBLIC_PHONE_2_DESC,
      })

    if (process.env.NEXT_PUBLIC_EMAIL_1_DESC)
      contactList.push({
        desc: process.env.NEXT_PUBLIC_EMAIL_1_DESC,
        link_external: process.env.NEXT_PUBLIC_EMAIL_1_DEVICE_LINK,
      })

    if (process.env.NEXT_PUBLIC_EMAIL_2_DESC)
      contactList.push({
        desc: process.env.NEXT_PUBLIC_EMAIL_2_DESC,
        link_external: process.env.NEXT_PUBLIC_EMAIL_2_DEVICE_LINK,
      })
  }

  return { customerData, menuData, contactList } as ICustomer
}

export default useCustomer

/**
 * Context
 */
export const CustomerContext = createContext<ICustomer>({} as ICustomer)

/**
 * Provider
 */
interface ICustomerProvider {
  initialValue?: ICustomer
}

export const CustomerProvider = ({
  children,
}: PropsWithChildren<ICustomerProvider>): React.ReactNode => {
  const { customerData, menuData, contactList } = useCustomer()
  console.log('DEBUG CustomerProvider > customerData:', customerData)

  // TODO: Return a loading component here.
  // return <h1>loading...</h1>

  return (
    <CustomerContext.Provider value={{ customerData, menuData, contactList }}>
      {children}
    </CustomerContext.Provider>
  )
}
