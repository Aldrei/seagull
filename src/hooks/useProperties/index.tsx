import { PropsWithChildren, createContext } from 'react'
import { IUseProperties } from './types'

export const useProperties = (): IUseProperties => {
  // Build URL after that make request from client...

  return { properties: [] }
}

export const PropertiesPage = createContext<IUseProperties>(
  {} as IUseProperties,
)

export const PropertiesProvider: React.FC<PropsWithChildren<{}>> = ({
  children,
}) => {
  const initialData = useProperties()

  return (
    <PropertiesPage.Provider value={{ ...initialData }}>
      {children}
    </PropertiesPage.Provider>
  )
}
