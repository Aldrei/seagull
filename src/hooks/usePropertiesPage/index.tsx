import { useRouter } from 'next/router'

export const usePropertiesPage = () => {
  const route = useRouter()

  const city = route.query['city-seo']
  const type = route.query['type-seo']
  const bed = route.query['bed-seo']

  console.log('DEBUG city:', city)
  console.log('DEBUG type:', type)
  console.log('DEBUG bed:', bed)

  return { filterOptions: city, type }
}
