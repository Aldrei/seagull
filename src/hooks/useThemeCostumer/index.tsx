import { TTheme } from '@/styles/themes/default/index.d'
import React, { createContext } from 'react'

// import { DefaultTheme, useTheme } from 'styled-components'

interface IProps {
  templateDefault: TTheme
}

const useThemeCostumer = ({ templateDefault }: IProps) => {
  // const templateDefault = useTheme()

  /**
   * TODO: Make the costumer client right below.
  */
 // Request...

 // Set colors right below...
 templateDefault.color.first = 'blue'

 const themeCostumer = templateDefault as TTheme

 return { themeCostumer }
}

export default useThemeCostumer

export const themeCostumerProvider = createContext(null)