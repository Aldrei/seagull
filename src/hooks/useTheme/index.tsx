import { TTheme } from '@/styles/themes/default/index.d'
import React, { PropsWithChildren, useContext } from 'react'

import { templateDefault } from '@/styles/themes/default'
import { CustomerContext } from '@/hooks/useCustomer'

import { ThemeProvider as ThemeProviderStyled } from 'styled-components'

/**
 * Hook
*/
interface IProps {
  templateDefault: TTheme
}

const useThemeCostumer = ({ templateDefault }: IProps) => {
  // const templateDefault = useTheme()

  /**
   * TODO: Make the costumer client right below.
  */
  // Request...

  // Set colors right below...
  templateDefault.color.first = 'blue'

  const themeCostumer = templateDefault as TTheme

  return { themeCostumer }
}

export default useThemeCostumer

/**
 * Provider
*/
interface IThemeProvider {}

export const ThemeProvider = ({ children }: PropsWithChildren<IThemeProvider>) => {
  // TODO: Make request to get data customer
  const customerData = useContext(CustomerContext)
  console.log('DEBUG customerData:', customerData);
  
  useThemeCostumer({ templateDefault })

  return (
    <ThemeProviderStyled theme={templateDefault}>{children}</ThemeProviderStyled>
  )
}
