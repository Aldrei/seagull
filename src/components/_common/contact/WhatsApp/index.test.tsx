import { render } from '@testing-library/react'

import { ThemeProvider } from '@/hooks/useTheme'

import { WhatsAppCommon } from './index'

describe('Common component', () => {
  it('Render WhatsAppCommon', () => {
    const el = render(
      <ThemeProvider>
        <WhatsAppCommon />
      </ThemeProvider>,
    )
    expect(el.baseElement).toMatchSnapshot()
  })
})
