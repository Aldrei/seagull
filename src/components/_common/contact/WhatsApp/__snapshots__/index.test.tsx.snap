// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common component Render WhatsAppCommon 1`] = `
<body>
  <div>
    <div
      class="sc-2865ade8-0 MMttG"
    >
      <div
        class="sc-2865ade8-1 hbYeXh"
      >
        <div
          class="sc-2865ade8-2 bhOnmI"
        >
          <img
            alt="Foto do corretor de imÃ³veis"
            class="sc-2865ade8-3 huzkqc"
          />
          <div
            class="sc-2865ade8-4 lhcFVZ"
          >
            <h2
              class="sc-2865ade8-5 lcBAOE"
            />
            <p
              class="sc-2865ade8-6 eGxYpV"
            >
              online
            </p>
          </div>
        </div>
        <div
          class="sc-2865ade8-7 jghLkA"
        >
          <a
            class="sc-2865ade8-8 bgnwey"
            target="_blank"
            title=""
          >
            Fale conosco
            <svg
              class="sc-2865ade8-9 eolila"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M476 3.2L12.5 270.6c-18.1 10.4-15.8 35.6 2.2 43.2L121 358.4l287.3-253.2c5.5-4.9 13.3 2.6 8.6 8.3L176 407v80.5c0 23.6 28.5 32.9 42.5 15.8L282 426l124.6 52.2c14.2 6 30.4-2.9 33-18.2l72-432C515 7.8 493.3-6.8 476 3.2z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;
